### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

File: src/main/java/mate/academy/dao/CarDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import mate.academy.db.Storage;
+import mate.academy.lib.Dao;
+import mate.academy.model.Car;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return Storage.cars.get(Storage.cars.size() - 1);
```
Comment (Baldinini) [inline src/main/java/mate/academy/dao/CarDaoImpl.java]: разве нельзя просто вернуть car?

File: src/main/java/mate/academy/dao/DriverDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import mate.academy.db.Storage;
+import mate.academy.lib.Dao;
+import mate.academy.model.Driver;
+
+@Dao
+public class DriverDaoImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        Storage.addDriver(driver);
+        return Storage.drivers.get(Storage.drivers.size() - 1);
```
Comment (Baldinini) [inline src/main/java/mate/academy/dao/DriverDaoImpl.java]: тоже самое

**Review by Baldinini — CHANGES_REQUESTED** [review]

**Review by Baldinini — APPROVED** [review]

File: src/main/java/mate/academy/dao/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return Storage.cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        Car oldValue = get(car.getId()).orElseThrow();
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/CarDaoImpl.java]:
```suggestion
        Car oldValue = get(car.getId()).get();
```

File: src/main/java/mate/academy/dao/CarDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        Storage.cars.remove(oldValue);
+        Storage.cars.add(car);
+        return oldValue;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/CarDaoImpl.java]: let's return new value

File: src/main/java/mate/academy/dao/CarDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.cars.removeIf(car -> Objects.equals(id, car.getId()));
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        return getAll().stream()
+                .filter(car -> car.getDrivers().contains(driverDao.get(driverId).orElseThrow()))
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/CarDaoImpl.java]: can we operate only with `driverId` without driverDao.get() method?

File: src/main/java/mate/academy/service/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.getDrivers().add(driver);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/CarServiceImpl.java:42]: we should call method update() after driver removing

File: src/main/java/mate/academy/service/CarServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().removeIf(x -> Objects.equals(driver, x));
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/CarServiceImpl.java]: same here

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**